jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/JibPluginConfiguration.java

```diff
@@ -145,14 +144,52 @@ static ImageReference parseImageReference(String image, String type) {
   @VisibleForTesting
   @Nullable
   static Authorization getImageAuth(
-      String usernameProperty, String passwordProperty, AuthConfiguration auth) {
+      JibLogger logger,
+      String imageProperty,
+      String usernameProperty,
+      String passwordProperty,
+      AuthConfiguration auth) {
     // System property takes priority over build configuration
     String commandlineUsername = System.getProperty(usernameProperty);
     String commandlinePassword = System.getProperty(passwordProperty);
     if (commandlineUsername != null && commandlinePassword != null) {
       return Authorizations.withBasicCredentials(commandlineUsername, commandlinePassword);
+    } else {
+      if (commandlinePassword != null) {
```

Comment:
I'm not sure how I feel about putting this logic in the configuration classes. I'm expecting we'll be refactoring the plugins pretty soon in preparation for the jib-plugins-common module though, so I decided to put it here until we have that structure figured out. However, I can move this somewhere else now if you want.

Reply:
Maybe where this authorization is used (ie. Build...Mojo where it makes the knownRegistryCredentials)? (So it would try to fetch and validate on it)

Reply:
Yeah, but there are 3 files in each plugin where that happens. I know we're going to be trying to reduce copy-paste soon, but it seemed like a little much.

Reply:
I think if we keep the actual snippet of code that does the knownRegistryCredential setup the same (with this added), it can actually help in the refactoring process so that we have one chunk of the same code to move and refactor to a common place (compared to trying to find this code hidden in the configuration class to factor away too).

Reply:
Ok, but we should refactor soon because this is going to be really bad.

Reply:
Yes, definitely, should be a top priority since the library depends on the frontend package being refactored away too

Reply:
Not following this, so are we good now, or will you make further changes in this PR?

Reply:
I think it should be good for review.